subsys/bootloader/c_runtime_setup/CMakeLists.txt

```diff
@@ -52,68 +40,8 @@ target_include_directories_ifdef(CONFIG_SB_C_RUNTIME_SETUP_VARIANT_ZEPHYR
   ${ARM_INC}/include/cortex_m
   )
 
-if(CONFIG_SB_C_RUNTIME_SETUP_VARIANT_ZEPHYR)
-  add_library(kernel_init STATIC ${ZEPHYR_BASE}/kernel/init.c)
-  target_link_libraries(kernel_init zephyr_interface)
-  target_compile_definitions(kernel_init PRIVATE _Cstart=UNUSED_Cstart)
-  add_dependencies(kernel_init offsets_h)
-  target_link_libraries(c_runtime_setup PRIVATE kernel_init)
-endif()
-
-# Nordic specific for startup code
-if(CONFIG_SOC_SERIES_NRF51X)
-target_sources_ifdef(CONFIG_SB_VENDOR_SYSTEM_INIT
-  c_runtime_setup
-  PRIVATE
-  ${HAL}/nordic/nrfx/mdk/system_nrf51.c
-  )
-target_sources_ifdef(CONFIG_SB_C_RUNTIME_SETUP_VARIANT_MDK
-  c_runtime_setup
-  PRIVATE
-  gcc_startup_nrf51.S
-  )
-elseif(CONFIG_SOC_NRF52810)
-target_sources_ifdef(CONFIG_SB_VENDOR_SYSTEM_INIT
-  c_runtime_setup
-  PRIVATE
-  ${HAL}/nordic/nrfx/mdk/system_nrf52810.c
-  )
-target_sources_ifdef(CONFIG_SB_C_RUNTIME_SETUP_VARIANT_MDK
-  c_runtime_setup
-  PRIVATE
-  gcc_startup_nrf52810.S
-  )
-elseif(CONFIG_SOC_NRF52832)
-target_sources_ifdef(CONFIG_SB_VENDOR_SYSTEM_INIT
-  c_runtime_setup
-  PRIVATE
-  ${HAL}/nordic/nrfx/mdk/system_nrf52.c
-  )
-target_sources_ifdef(CONFIG_SB_C_RUNTIME_SETUP_VARIANT_MDK
-  c_runtime_setup
-  PRIVATE
-  gcc_startup_nrf52.S
-  )
-elseif(CONFIG_SOC_NRF52840)
-target_sources_ifdef(CONFIG_SB_VENDOR_SYSTEM_INIT
-  c_runtime_setup
-  PRIVATE
-  ${HAL}/nordic/nrfx/mdk/system_nrf52840.c
-  )
-target_sources_ifdef(CONFIG_SB_C_RUNTIME_SETUP_VARIANT_MDK
-  c_runtime_setup
-  PRIVATE
-  gcc_startup_nrf52840.S
-  )
-elseif(CONFIG_SOC_NRF9160)
-target_sources_ifdef(CONFIG_SB_VENDOR_SYSTEM_INIT
-  c_runtime_setup
-  PRIVATE
-  ${HAL}/nordic/nrfx/mdk/system_nrf9160.c
-  )
-target_sources_ifdef(CONFIG_SB_C_RUNTIME_SETUP_VARIANT_MDK
-  c_runtime_setup
-  PRIVATE
-  gcc_startup_nrf9160.S
-  )
-endif()
+add_library(kernel_init STATIC ${ZEPHYR_BASE}/kernel/init.c)
```

Comment:
This isn't a zephyr library, because it should be linked into the bootloader, and not the app, right?

Please add a comment that explains this.

Then I wouldn't be confused again, and it will be more clear that this needs to change after the multi-image migration.

Reply:
Yes, you are correct it isn't a zephyr library it was a hackish way of being able to set `_Cstart=UNUSED_Cstart` only for compiling the C runtime/startup code of zephyr then override the _Cstart rutine with our bootloaders `main_bl` call. I'll add a comment explaining this. Do you mind elaborating a bit on what needs to be changed after multi-image ?

Reply:
Added comment @SebastianBoe check if the comment gives enough information so that you are not confused by the purpose of this lib.

Reply:
LGTM